src/sentry/static/sentry/app/views/organizationDiscover/queryBuilder.jsx

```diff
@@ -184,6 +184,10 @@ export default function createQueryBuilder(initial = {}, organization) {
       }
     }
 
+    if (moment.utc(data.start).isAfter(moment.utc(data.end))) {
+      return Promise.reject(new Error('Invalid date range'));
```

Comment:
Should you tell the user that their start date must be before the end date?

Reply:
Done